substrate/frame/nfts/src/impl_nonfungibles.rs

```diff
@@ -320,6 +320,33 @@ impl<T: Config<I>, I: 'static> Mutate<<T as SystemConfig>::AccountId, ItemConfig
 		})
 	}
 
+	fn set_metadata(
+		who: &T::AccountId,
```

Comment:
I think we should remove that `who` param and follow the same pattern we set the attributes by passing `None` into the `do_*` method

Reply:
Or we can make it optional :) maybe this would even be a more universal approach

Reply:
```
	fn set_metadata(
		who: Option<&T::AccountId>,
		collection: &Self::CollectionId,
		item: &Self::ItemId,
		data: &[u8],
	) -> DispatchResult {
		Self::do_set_item_metadata(
			who.clone(),
			*collection,
			*item,
			Self::construct_metadata(data.to_vec())?,
			None,
		)
	}
```

Reply:
Done. I used the optional approach